services/repository/archiver/archiver.go

```diff
@@ -251,13 +252,23 @@ func doArchive(r *ArchiveRequest) (*repo_model.RepoArchiver, error) {
 				w,
 			)
 		} else {
-			err = gitRepo.CreateArchive(
-				ctx,
-				archiver.Type,
-				w,
-				setting.Repository.PrefixArchiveFiles,
-				archiver.CommitID,
-			)
+			if annex.IsAnnexRepo(gitRepo) {
+				err = annex.RepoArchive(
+					gitRepo,
+					archiver.Type,
+					w,
```

Comment:
This probably needs `ctx`, to support cancellation, the same way `git.CreateArchive()` does, like if the user cancels the download.

Also it'd be good to keep close to upstream here in the naming!

```suggestion
				err = annex.CreateArchive(
				  ctx,
					gitRepo,
					archiver.Type,
					w,
```

Reply:
You are right, will update